kubernetes/structure_service_spec.go

```diff
@@ -167,6 +167,13 @@ func patchServiceSpec(keyPrefix, pathPrefix string, d *schema.ResourceData, v *v
 		})
 	}
 	if d.HasChange(keyPrefix + "type") {
+		o, n := d.GetChange(keyPrefix + "type")
+
+		if o.(string) == "ClusterIP" && n.(string) == "ExternalName" {
```

Comment:
This is great! It seems to work reliably, and I like that there's a test for it too. I found some other cases where this logic will be useful. Here's a summary of what I found during my testing:

1. If cluster_ip exists in state already, and we're trying to change the service to type ExternalName, we need to remove cluster_ip. ClusterIP is automatically set by `type: NodePort`, `type: ClusterIP`, and `type: LoadBalancer`. So basically we need to expand the current logic in this PR to handle those types as well.

2. If `spec.ports[0].nodePort` is set, and we're trying to change the service type to ClusterIP, we need to delete nodePort. This spec.[0].nodePort is automatically created by `type: NodePort` and `type: LoadBalancer`, and it can be used with ExternalName. Users can specify this with any type other than `ClusterIP`.

Let me know if I should create a new bug report for number 2. That could be more trouble than it's worth in this PR, so I'll leave that decision to you.

Here's how I think we can cover number 1:

```suggestion
		if n.(string) == "ExternalName" {
```

Reply:
Thanks for the feeback, first issue makes sense ill test it. for 2 ill add the logic and test to see but no need to wait i think. if i wont respond in a timely manner feel free to merge this and open a new PR.

Reply:
Ok, i think i managed to solve the second case as well. not sure this is the nicest solution but this what i was able to cook up quickly. LMK what you think.

Reply:
acctest after latest change:

```
--- PASS: TestAccKubernetesService_basic (7.09s)
--- PASS: TestAccKubernetesService_nodePort (4.72s)
--- PASS: TestAccKubernetesService_externalName (2.50s)
--- PASS: TestAccKubernetesService_externalName_toClusterIp (7.16s)
--- PASS: TestAccKubernetesService_generatedName (3.15s)
--- PASS: TestAccKubernetesService_regression (161.28s)
```